app/service/rest.py

```diff
@@ -210,6 +214,98 @@ def get_service_notification_statistics(service_id):
     )
 
 
+@service_blueprint.route("/<uuid:service_id>/statistics/<string:start>/<int:days>")
+def get_service_notification_statistics_by_day(service_id, start, days):
+    return jsonify(
+        data=get_service_statistics_for_specific_days(service_id, start, int(days))
+    )
+
+
+def get_service_statistics_for_specific_days(service_id, start, days=1):
+    start_date = datetime.strptime(start, "%Y-%m-%d").date()
+
+    def generate_date_range(start_date, days):
+        current_date = start_date
+        end_date = start_date - timedelta(days=days)
+        while current_date > end_date:
+            try:
+                valid_date = datetime(
+                    current_date.year, current_date.month, current_date.day
+                )
+                yield valid_date.date()
+            except ValueError:
+                pass
+            current_date -= timedelta(days=1)
+
+    results = dao_fetch_stats_for_service_from_days(service_id, start_date, days)
+
+    grouped_results = defaultdict(list)
+    for row in results:
+        notification_type, status, day, count = row
+        grouped_results[day.date()].append(row)
+
+    for date in generate_date_range(start_date, days):
+        if date not in grouped_results:
+            grouped_results[date] = []
+
+    stats = {}
+    for day, rows in grouped_results.items():
+        stats[day.strftime("%Y-%m-%d")] = statistics.format_statistics(rows)
+
+    return stats
+
+
+@service_blueprint.route(
+    "/<uuid:service_id>/statistics/user/<uuid:user_id>/<string:start>/<int:days>"
+)
+def get_service_notification_statistics_by_day_by_user(
+    service_id, user_id, start, days
+):
+    return jsonify(
+        data=get_service_statistics_for_specific_days_by_user(
+            service_id, user_id, start, int(days)
+        )
+    )
+
+
+def get_service_statistics_for_specific_days_by_user(
+    service_id, user_id, start, days=1
+):
+    start_date = datetime.strptime(start, "%Y-%m-%d").date()
+
+    def generate_date_range(start_date, days):
```

Comment:
This is a copy/paste from above?  Consider making generate_date_range a standalone method and reusing it.

Reply:
I can move it into the utilities.